### ISSUE TITLE
Cannot start king-phisher.service

### ISSUE THREAD

**Issue by anhp11** [securestate/king-phisher#119]
## Issue Description
I'm trying to install and set up King-phisher on Ubuntu 16.04 LTS and was able to install successfully with all requirements. However, when attempted to start king-phisher.service, I got the following error.

"Job for king-phisher.service failed because a timeout was exceeded. See "systemctl status king-phisher.service" and "journalctl -xe" for details"

## Environment Details
Host OS: Ubuntu 16.04 LTS

## Error Details / Stack Trace

```
root@lbkingphisher:/opt/king-phisher/tools# systemctl status king-phisher.service
● king-phisher.service - King Phisher Server
   Loaded: loaded (/lib/systemd/system/king-phisher.service; enabled; vendor preset: enabled)
   Active: failed (Result: timeout) since Mon 2016-06-13 11:12:56 EDT; 28min ago
  Process: 6612 ExecStart=/opt/king-phisher/KingPhisherServer /opt/king-phisher/server_config.yml (code=exited, status=0/SUCCESS)

Jun 13 11:11:27 lbkingphisher KingPhisherServer[6612]:     self._db_reader = maxminddb.open_database(filename, mode)
Jun 13 11:11:27 lbkingphisher KingPhisherServer[6612]:   File "/usr/local/lib/python2.7/dist-packages/maxminddb/__init__.py", line 34, in open_database
Jun 13 11:11:27 lbkingphisher KingPhisherServer[6612]:     return maxminddb.reader.Reader(database, mode)
Jun 13 11:11:27 lbkingphisher KingPhisherServer[6612]:   File "/usr/local/lib/python2.7/dist-packages/maxminddb/reader.py", line 53, in __init__
Jun 13 11:11:27 lbkingphisher KingPhisherServer[6612]:     db_file.fileno(), 0, access=mmap.ACCESS_READ)
Jun 13 11:11:27 lbkingphisher KingPhisherServer[6612]: ValueError: cannot mmap an empty file
Jun 13 11:12:56 lbkingphisher systemd[1]: king-phisher.service: Start operation timed out. Terminating.
Jun 13 11:12:56 lbkingphisher systemd[1]: Failed to start King Phisher Server.
Jun 13 11:12:56 lbkingphisher systemd[1]: king-phisher.service: Unit entered failed state.
Jun 13 11:12:56 lbkingphisher systemd[1]: king-phisher.service: Failed with result 'timeout'.

```

**Comment by zeroSteiner** [comment]
It's complaining that the GeoIP database file is empty. Try deleting it (find the location from the config file) and restarting the server. It will be re-downloaded to when the server starts.

**Comment by zeroSteiner** [comment]
I've marked this as a bug since an empty database file should just cause the server to download a new one. I've also implemented this change already in the dev branch.

I'll leave this ticket open though for a while longer while I wait for your confirmation of the work around I described in my previous post.

**Comment by anhp11** [comment]
That worked!!! Thank you. After I removed the file and restarted the server, I got this error while restarting

**root@lbkingphisher:/var/king-phisher# systemctl status king-phisher.service
● king-phisher.service - King Phisher Server
   Loaded: loaded (/lib/systemd/system/king-phisher.service; enabled; vendor preset: enabled)
   Active: failed (Result: resources) since Wed 2016-06-15 06:49:07 EDT; 4s ago
  Process: 16219 ExecStart=/opt/king-phisher/KingPhisherServer /opt/king-phisher/server_config.yml (code=exited, status=0/SUCCESS)

Jun 15 06:49:07 lbkingphisher systemd[1]: Starting King Phisher Server...
Jun 15 06:49:07 lbkingphisher KingPhisherServer[16219]: CRITICAL failed to bind server to address 0.0.0.0:80 (socket error #98)
Jun 15 06:49:07 lbkingphisher KingPhisherServer[16219]: CRITICAL socket error #98 (Address already in use)
Jun 15 06:49:07 lbkingphisher systemd[1]: king-phisher.service: PID file /var/run/king-phisher.pid not readable (yet?) after start: No such file or directory
Jun 15 06:49:07 lbkingphisher systemd[1]: king-phisher.service: Daemon never wrote its PID file. Failing.
Jun 15 06:49:07 lbkingphisher systemd[1]: Failed to start King Phisher Server.
Jun 15 06:49:07 lbkingphisher systemd[1]: king-phisher.service: Unit entered failed state.
Jun 15 06:49:07 lbkingphisher systemd[1]: king-phisher.service: Failed with result 'resources'.**

Then I found this when doing a netstat with grepping on port 80: 

**tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      12042/python**   

After killing that process manually, I was able to start King-Phisher successfully. However, I was wondering if that was installed and set up by running the installs.sh script King-phisher provided, if so, that might need to be changed.

Thanks again!
